boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,373 @@
+ * set/clear the parity bit. Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit set/cleared
+ */
+static void Io_Efuse_CalculateParityBit(uint16_t *serial_data_input);
+
+/**
+ * Write SPI data to Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param efuse_spi_handle Handle to the Efuse's SPI peripheral
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_WriteToEfuse(
+    uint16_t *         tx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+/**
+ * Read SPI data from Efuse in blocking mode
+ * @param tx_data Pointer to data being sent to the Efuse
+ * @param rx_data Pointer to data being received from the Efuse
+ * @param efuse_spi_handle Handle to the Efuse's SPI peripheral
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *const hspi,
+    GPIO_TypeDef *           chip_select_port,
+    uint16_t                 chip_select_pin,
+    GPIO_TypeDef *           fsob_port,
+    uint16_t                 fsob_pin,
+    GPIO_TypeDef *           fsb_port,
+    uint16_t                 fsb_pin,
+    GPIO_TypeDef *           channel0_port,
+    uint16_t                 channel0_pin,
+    GPIO_TypeDef *           channel1_port,
+    uint16_t                 channel1_pin)
+{
+    assert(hspi != NULL);
+
+    struct Efuse_Context *efuse_context = malloc(sizeof(struct Efuse_Context));
+    assert(efuse_context != NULL);
+
+    efuse_context->hspi            = hspi;
+    efuse_context->nss_port        = chip_select_port;
+    efuse_context->nss_pin         = chip_select_pin;
+    efuse_context->fsob_port       = fsob_port;
+    efuse_context->fsob_pin        = fsob_pin;
+    efuse_context->fsb_port        = fsb_port;
+    efuse_context->fsb_pin         = fsb_pin;
+    efuse_context->channel0_port   = channel0_port;
+    efuse_context->channel0_pin    = channel0_pin;
+    efuse_context->channel1_port   = channel1_port;
+    efuse_context->channel1_pin    = channel1_pin;
+    efuse_context->wdin_bit_to_set = true;
+
+    return efuse_context;
+}
+
+void Io_Efuse_Destroy(struct Efuse_Context *e_fuse)
+{
+    free(e_fuse);
```

Comment:
Will we ever run into a situation where this fxn is needed?

Reply:
Would make sense if it was in the App layer. I actually don't think we need to free structs allocated in the Io level.

Reply:
probably not, I was just following the convention with other files